src/rules/prefer-native-locators.ts

```diff
@@ -0,0 +1,139 @@
+import { getStringValue, isPageMethod } from '../utils/ast'
+import { createRule } from '../utils/createRule'
+
+export default createRule({
+  create(context) {
+    return {
+      CallExpression(node) {
+        const { testIdAttribute } = {
+          testIdAttribute: 'data-testid',
+          ...((context.options?.[0] as Record<string, unknown>) ?? {}),
+        }
+
+        if (node.callee.type !== 'MemberExpression') return
+        const method = getStringValue(node.callee.property)
+        const query = getStringValue(node.arguments[0])
+        const isLocator = isPageMethod(node, 'locator') || method === 'locator'
+        if (!isLocator) return
+
+        // If it's something like `page.locator`, just replace the `.locator` part
+        const start =
+          node.callee.type === 'MemberExpression'
+            ? node.callee.property.range![0]
+            : node.range![0]
+        const end = node.range![1]
+        const rangeToReplace: [number, number] = [start, end]
+
+        const ariaLabelPattern = /^\[aria-label=['"](.+?)['"]\]$/
```

Comment:
Lot's of duplicated code, let's make a list of regexes and their corresponding built-in locators and just loop over it.

Reply:
Good idea, I've replaced this with a fairly simple array of patterns plus other data needed to perform the replacement. Let me know what you think: https://github.com/playwright-community/eslint-plugin-playwright/pull/308/commits/61fbebb476281c163e1c1a81b2a258259bdc3f9c